src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx

```diff
@@ -24,12 +25,30 @@ export default function QuoteMessageThumbnail({ message }: Props): ReactElement
   if (!isFileMessage(message) && !isMultipleFilesMessage(message) || isVoiceMessage(message as FileMessage)) {
     return null;
   }
-  const thumbnailUrl: string = isFileMessage(message)
-    ? (getMessageFirstFileThumbnailUrl(message) || (isImageMessage(message as FileMessage) && getMessageFirstFileUrl(message)))
-    : (getMessageFirstFileThumbnailUrl(message)
-      || isImageFileInfo((message as MultipleFilesMessage).fileInfoList[0])
-      && getMessageFirstFileUrl(message));
-  if ((isThumbnailMessage(message) || isMultipleFilesMessage(message)) && thumbnailUrl) {
+  const thumbnailUrl: string = match(message)
+    .when(isFileMessage, () => {
+      return getMessageFirstFileThumbnailUrl(message)
+        || (
+          (
+            isImageMessage(message as FileMessage)
+            || isVideoMessage(message as FileMessage)
+          )
+          && getMessageFirstFileUrl(message)
+        );
+    })
+    .when(isMultipleFilesMessage, () => {
+      const castedMessage: MultipleFilesMessage = message as MultipleFilesMessage;
+      return getMessageFirstFileThumbnailUrl(castedMessage)
+        || (
+          castedMessage.fileInfoList.length > 0
+          && isImageFileInfo((message as MultipleFilesMessage).fileInfoList[0])
+          && getMessageFirstFileUrl(message)
+        );
+    })
+    .otherwise(() => {
+      return getMessageFirstFileThumbnailUrl(message);
+    });
+  if (!isVideoMessage(message) && (isThumbnailMessage(message) || isMultipleFilesMessage(message)) && thumbnailUrl) {
```

Comment:
How about this?

```tsx
  let thumbnailUrl = getMessageFirstFileThumbnailUrl(message);
  if (!thumbnailUrl && message.isFileMessage() && isImageMessage(message)) {
    thumbnailUrl = getMessageFirstFileUrl(message);
  }
  if (!thumbnailUrl && message.isMultipleFilesMessage() && isImage(message.fileInfoList?.[0].mimeType)) {
    thumbnailUrl = message.fileInfoList[0].url;
  }
  
  if (thumbnailUrl) {
    return (
      <ImageRenderer
        className={componentClassname}
        url={thumbnailUrl}
        alt={getMessageFirstFileType(message)}
        width="44px"
        height="44px"
        fixedSize
      />
    );
  }
```

Reply:
better 👍

Reply:
how come copying the code snippet makes paste into a single line? 😢